pe.c

```diff
@@ -1277,8 +1277,11 @@ handle_image (void *data, unsigned int datasize,
 					Section->Misc.VirtualSize &&
 					base && end &&
 					RelocBase == base &&
-					RelocBaseEnd == end) {
+					RelocBaseEnd <= end) {
```

Comment:
Without this change, the file loads but crashes (since shim will attempt to use the file as if there is no reloc section when there is).

Reply:
Hi,

Thanks for this input.

Is BS->FreePages correct, here? The various other (pre-existing) `return EFI_UNSUPPORTED` nearby (and in the same allocation context, I believe), do not do it.

Then on the general idea, yes, in the current version of this PR the code continues if no reloc section is finally found (because this is what the code previously did), but the correct place to decide not to do so (which I tend to agree would be an improvement) would be around line 1331: https://github.com/rhboot/shim/blob/0640e136842200eb5873e9b2b02b159e15940591/pe.c#L1331

If you (possibly) meant the suggestion as an alternative fix - i.e. to avoid my crash - then I think that misunderstands the nature of the PR - the file _should_ load and run (not abort, and also - ofc - not crash) and this PR allows it to load.